x-pack/plugins/observability_solution/entity_manager/server/lib/manage_index_templates.ts

```diff
@@ -37,6 +43,15 @@ export async function upsertTemplate({ esClient, template, logger }: TemplateMan
   logger.debug(() => `Entity manager index template: ${JSON.stringify(template)}`);
 }
 
+export async function deleteTemplate({ esClient, name, logger }: DeleteTemplateOptions) {
+  try {
+    await esClient.indices.deleteIndexTemplate({ name });
```

Comment:
I think this would fail if the index template does not exist ? if so could we ignore 404 errors like [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/entity_manager/server/lib/entities/delete_ingest_pipeline.ts#L24)

Also can we wrap the methods in this file with `retryTransientEsErrors` like the ingest pipeline example above ?